### PR TITLE
Effects during battle

### LINKED ISSUE #50: effects during battle
when battle starts add gif at the center (few seconds) with two swords fighting

change music to battle sound

show number of enemies

### PR DESCRIPTION
**Pull request by hoolek77**

### CODE REVIEW

File: src/app/app.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export class App {
 
   showGamePage() {
     if (!this.gamePage) {
-      this.gamePage = new GamePage(this, this.game)
+      this.gamePage = new GamePage(this, this.game, this.audio)
```
Comment (pdybowski) [inline src/app/app.ts]: Hmm im not sure but 'this' contains game and audio so those two additional args are not necessary

File: src/app/screens/gamepage/index.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import { App } from '../../app'
+import { Audio } from '../../audio'
 import { GameOverModal } from '../../components'
+import { Battle } from '../../components'
```
Comment (pdybowski) [inline src/app/screens/gamepage/index.ts]: See line number 3

File: src/index.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ import './core/bootstrap'
 
 import { App } from './app'
 
+const app = new App()
```
Comment (pdybowski) [inline src/index.ts]: Why?

File: src/app/components/battle.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      this.mainArea.appendChild(this.battleContainer)
+      this.infoHeading.style.opacity = '0'
+      this.progress.style.opacity = '0'
+      this.createBattle()
+      this.audio.changeAudioSource('../../assets/audio/battle.mp3')
+    }
+  }
+
+  private removeBattle() {
+    if (this.infoHeading.style.opacity === '0') {
```
Comment (pdybowski) [inline src/app/components/battle.ts]: Can we set '==' here? It should be string but who knows what will happen in the future

File: src/app/components/battle.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import { createElement } from '../utils/createElement'
+import { Audio } from '../audio'
+import { Game } from '../game'
+
+export class Battle {
+  private audio: Audio
+  private game: Game
+  private mainArea: HTMLElement = document.querySelector(
```
Comment (pdybowski) [inline src/app/components/battle.ts]: Can we place all those elements(not Audio and Game) in _init() / getUIElements() or something like that and run this method in the constructor?

File: src/app/screens/gamepage/index.ts [hunk 6]
```diff
@@ -22,9 +25,10 @@ export class GamePage {
 
   private isGameOverModalVisible = false
 
-  constructor(app: App, game: Game) {
+  constructor(app: App, game: Game, audio: Audio) {
     this.app = app
     this.game = game
+    this.audio = audio
```
Comment (pdybowski) [inline src/app/screens/gamepage/index.ts]: If you follow my first comment here you can change it to this.audio = app.audio

**Review by pdybowski — CHANGES_REQUESTED** [review]
It works but you can make some improvements

**Review by pdybowski — APPROVED** [review]

File: src/app/screens/gamepage/index.ts [hunk 7]
```diff
@@ -79,6 +82,10 @@ export class GamePage {
       '.defence__count'
     ) as HTMLElement
 
+    this.progressHeader = document.querySelector(
```
Comment (bartoszbialecki) [inline src/app/screens/gamepage/index.ts]: Is it correct? Those elements are not definied below?

**Review by bartoszbialecki — APPROVED** [review]

**Pull request merged**